ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockCreator.java

```diff
@@ -195,8 +203,24 @@ protected BlockCreationResult createBlock(
 
       throwIfStopped();
 
-      final Optional<List<Deposit>> maybeDeposits =
-          Optional.empty(); // TODO 6110: Extract deposits from transaction receipts
+      final DepositsValidator depositsValidator = newProtocolSpec.getDepositsValidator();
+      Optional<List<Deposit>> maybeDeposits = Optional.empty();
+
+      if (depositsValidator instanceof DepositsValidator.AllowedDeposits) {
+        final List<Deposit> depositsFromReceipts =
+            transactionResults.getReceipts().stream()
+                .flatMap(receipt -> receipt.getLogsList().stream())
+                .filter(
+                    log ->
+                        depositContractAddress
+                            .orElse(DEFAULT_DEPOSIT_CONTRACT_ADDRESS)
```

Comment:
I don't think we should handle the default value in this class.

I think in this case it's fine to use orElseThrow since we shouldn't get into this case for non-mainnet usecases.
We'll probably need to retest with non-mainnet networks to make sure, I'm happy to do this testing.

Reply:
...in order for this error to be thrown I think the genesis file would need to have `experimentalEipsTime` configured and be missing some `depositContractAddress` configuration. I think this is acceptable for this PR.

Me and @jframe discussed a refactoring of this class to move the withdrawals and deposit code into MergeBlockCreator (which should maybe now be called 'ProofOfStakeBlockCreator') since withdrawals and deposits are specific to mainnet/PoS.
I think that refactoring would reduce the risk of misconfiguration further. 

Maybe a default is still the way to go but it should be as close to when the config is read-in rather than when it is used...maybe @jframe has an opinion on this too.

Reply:
Agree that @siladu that we shouldn't have a default defined. We can assume that if this EIP is enabled then we will have a deposit contract defined in the relevant network genesis file.

Reply:
That refactoring to move the mainnet specific features out of the AbstractBlockCreator would be nice to have but I think that is something for the future that could be done in another PR.